### PR TITLE
Do not crash with a traceback if a machine value is unset

### PR DESCRIPTION
**Pull request by frankban**
Handle the case in which the machines section includes a machine without a definition, like:
```
machines:
    1:
```
Both validation and changeset generation crashed with a YAML like that.
The tests should demonstrate the new behavior.

Please `make check`.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-bundlelib/44/
Test PASSed.

**Comment by makyo** [comment]
:+1: thanks for the fix!

File: jujubundlelib/validation.py [hunk 1]
```diff
@@ -330,12 +330,17 @@ def _validate_placement(placement, services, machines, charm, add_error):
...
+        # A machine can be included in machines but its value can be None.
+        # This is so that we are compatible with go-style YAML unmarshaling.
+        if machine_id not in machines:
             add_error(
                 'placement {} refers to a non-existent machine {}'
                 ''.format(placement, unit_placement.machine))
             return
+        machine = machines[machine_id]
+        if not isdict(machine):
+            # This error is notified while validating machines.
```
Comment (bac) [inline jujubundlelib/validation.py]: Could you rephrase that sentence? What do you mean by 'is notified'?

**Comment by bac** [comment]
:+1:

File: jujubundlelib/validation.py (hunk 1 above)
Comment (frankban) [inline jujubundlelib/validation.py]: Done.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-bundlelib/45/
Test PASSed.

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-bundlelib-merge
